test/unit/template-parts.spec.js

```diff
@@ -0,0 +1,579 @@
+  is(template.innerHTML, originalHTML);
+  const root = document.createElement('div');
+  root.appendChild(instance);
+  is(root.innerHTML, `<div class="foo"></div>`);
+});
+
+test('create: can render into partial attribute nodes', () => {
+  const template = document.createElement('template');
+  const originalHTML = `<div class="my-{{y}}-state"></div>`;
+  template.innerHTML = originalHTML;
+  const instance = new TemplateInstance(template, { y: 'foo' });
+  is(template.innerHTML, originalHTML);
+  const root = document.createElement('div');
+  root.appendChild(instance);
+  is(root.innerHTML, `<div class="my-foo-state"></div>`);
+});
+
+test('create: can render into many values', () => {
+  const template = document.createElement('template');
+  const originalHTML = `<div class="my-{{x}}-state {{y}}">{{z}}</div>`;
+  template.innerHTML = originalHTML;
+  const instance = new TemplateInstance(template, {
+    x: 'foo',
+    y: 'bar',
+    z: 'baz',
+  });
+  is(template.innerHTML, originalHTML);
+  const root = document.createElement('div');
+  root.appendChild(instance);
+  is(root.innerHTML, `<div class="my-foo-state bar">baz</div>`);
+});
+
+test('create: it allows spaces inside template part identifiers', () => {
+  const template = document.createElement('template');
+  const originalHTML = `<div class="my-{{ x }}-state {{ y }}">{{         z          }}</div>`;
+  template.innerHTML = originalHTML;
+  const instance = new TemplateInstance(template, {
+    x: 'foo',
+    y: 'bar',
+    z: 'baz',
+  });
+  is(template.innerHTML, originalHTML);
+  const root = document.createElement('div');
+  root.appendChild(instance);
+  is(root.innerHTML, `<div class="my-foo-state bar">baz</div>`);
+});
+
+test('create: never writes mustache syntax into an instantiated template even if no state given', () => {
+  const template = document.createElement('template');
+  const originalHTML = `<div class="my-{{ x }}-state {{ y }}">{{ z }}</div>`;
+  template.innerHTML = originalHTML;
+  const instance = new TemplateInstance(template);
+  is(template.innerHTML, originalHTML);
+  const root = document.createElement('div');
+  root.appendChild(instance);
+  is(root.innerHTML, `<div class="my--state "></div>`);
+});
+
+test('update: updates all nodes with new values', () => {
+  const template = document.createElement('template');
+  const originalHTML = `<div class="my-{{ x }}-state {{ y }}">{{ z }}</div>`;
+  template.innerHTML = originalHTML;
+  const instance = new TemplateInstance(template, {
+    x: 'foo',
+    y: 'bar',
+    z: 'baz',
+  });
+  is(template.innerHTML, originalHTML);
+  const root = document.createElement('div');
+  root.appendChild(instance);
+  is(root.innerHTML, `<div class="my-foo-state bar">baz</div>`);
+  instance.update({ x: 'bing', y: 'bong', z: 'quux' });
+  is(root.innerHTML, `<div class="my-bing-state bong">quux</div>`);
+});
+
+test('update: performs noop when update() is called with partial args', () => {
```

Comment:
praise: another really good test to include for a spec, especially since it's not "intuitive" that it would behave this way.